### PR TITLE
Use distroless image in docker distribution

### PR DESCRIPTION
**Pull request by avelino**
fixed: #447

### CODE REVIEW

**Comment by felipeweb** [comment]
done @avelino  feel free to merge!

File: etc/entrypoint.sh [hunk 1]
```diff
@@ -1,4 +1,15 @@
-#!/usr/bin/env sh
```
Comment (avelino) [inline etc/entrypoint.sh:1]: depend on target shell and yours is unknown on shell script
ref: https://github.com/prest/prest/issues/449

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
-RUN apk add --no-cache curl
-COPY ./cmd/prestd/prest.toml /app/prest.toml
-COPY ./etc/entrypoint.sh /app/entrypoint.sh
-ENTRYPOINT [ "/app/entrypoint.sh" ]
+FROM golang:1.15 as builder
+
+WORKDIR /workspace
+COPY . .
+WORKDIR /workspace/cmd/prestd
+RUN go mod download
```
Comment (avelino) [inline Dockerfile:6]: group `RUN` commands, to reduce the number of docker layers

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
-RUN apk add --no-cache git && \
-        go mod tidy && go build
-
-FROM registry.hub.docker.com/library/alpine:latest
-COPY --from=builder /go/src/github.com/prest/prest/cmd/prestd/prestd /app/prestd
-RUN apk add --no-cache curl
-COPY ./cmd/prestd/prest.toml /app/prest.toml
-COPY ./etc/entrypoint.sh /app/entrypoint.sh
-ENTRYPOINT [ "/app/entrypoint.sh" ]
+FROM golang:1.15 as builder
```
Comment (avelino) [inline Dockerfile:1]: Use only an allowed registry in the FROM image
ref: https://github.com/prest/prest/issues/431

File: Dockerfile [hunk 4]
```diff
@@ -1,13 +1,25 @@
...
+
+RUN apt-get update && apt-get install -yq netcat
+
+WORKDIR /app
+
+
+# use debug because we need a shell (busybox)
+FROM gcr.io/distroless/base:debug 
+COPY --from=builder /bin/nc /bin/nc
+COPY --from=builder --chown=nonroot:nonroot  /app /app
```
Comment (avelino) [inline Dockerfile:19]: very good to set not to be root, congratulations 😃

**Comment by avelino** [comment]
more issues: https://app.codacy.com/gh/prest/prest/pullRequest?bid=20447309&prid=6283601&utm_campaign=new_commit&utm_medium=Email&utm_source=Internal&

File: etc/entrypoint.sh (hunk 1 above)
Comment (felipeweb) [reply etc/entrypoint.sh:1]: busybox shell do not parse arguments so I change to #!/bin/sh

**Comment by felipeweb** [comment]
sone @avelino

**Review by felipeweb — APPROVED** [review]

**Pull request merged**